ls_dev

Nabegon opened the pull request:
lsコマンドの課題を提出いたします。
ご確認のほどよろしくお願いします。

Nabegon committed:
first commit

Nabegon committed:
pass lint using rubocop

Nabegon commented:
option_l()メソッドが長すぎてlintが通らず修正。
冗長な変数への代入を削除しました。

JunichiIto commented:
動画でレビューしたので別途リンクを送ります〜。

Nabegon committed:
fix wrong last access time

DateTime class is now used to get the correct last access time

Nabegon committed:
fix: use Dir.glob to get and ckeck file names

Dir.glob is now used to get file names and see if names start with . instead of using Dir.foreach and checking file names by myself

Nabegon committed:
remove: dupilicated reverse methods

Used two times reverse methods in get_files() and print_file_details(). the method is now used only in get_files(). For that, the list of file names is sortedalso in get_files(), not in option_l().

Nabegon committed:
fix: multi column layout

each_slice method is used for displaying the retrived fine names in up to three columns, instead of writing code by myself.

Nabegon committed:
fix: use constants for get_file_type() and get_permission()

for file type and file permissions, constants are now used instead of variables because they use initial values. Classes are used to define constants.

Nabegon committed:
remove: classes for constants

constants for file type and file permissions are defined outside of methods, without using classes.

Nabegon committed:
remove: duplicated join method

Nabegon committed:
fix: inside show_files(), process to display files in three columns

fixed some hard-coded parts to make the code more resistant to changes. Also modified the method which displays files, it now uses a double loop to display the elements in a two-dimensional array.

Nabegon committed:
rename: methods and variables

Nabegon commented:
> 動画でレビューしたので別途リンクを送ります〜。

お疲れ様です。レビュー後修正いたしました。再度ご確認のほどよろしくお願いします。
修正点は主に下記の通りです。
・ファイルの最終アクセス日時を正しく取得（月が正しく取得できていなかった）
・ファイルの取得はDir.globを使い、ドット付きファイルを取得する場合はDir.globのglob()の引数内で指定
・join, reverseが使わずに2回書かれているところがあり、そちらを1つにまとめてメソッドも書き直しています
・3列表示のメソッドのブロック構造を修正し、ハードコードされていた部分を変更に強いコードに修正
・全体的にメソッドと変数の名前を変更

以上になります。よろしくお願いします。

Nabegon committed:
fix: handling different array sizes and processing when a directory is empty

used reduce(&:zip) to handle different size arrays but now each do block is used. Also directory is now checked and if it's empty, there is a new processing code to exit the entire process.'

JunichiIto commented:
今回も動画でレビューしたので別途リンクを送ります〜。

Nabegon committed:
remove: error message and exit methods

removed exit methods and error message when a non-existent path or option is given or an empty directory is given.

Nabegon committed:
modified after 2nd review

JunichiIto reviewed (commented):
またまた動画でレビューしてみました〜。

Nabegon committed:
modified after 3rd review

JunichiIto commented on 05.ls/ls.rb:
```diff
@@ -0,0 +1,105 @@
+
+def print_file_matrix(files)
+  file_matrix = build_file_matrix(files)
+  max_length_string = file_matrix.flatten.compact.max_by(&:length).length
```
`max_length_string`だと「一番長い文字列」の意味になるので、中身が文字列であると勘違いされそうです。
数字であることを示すなら、`max_length`とか`max_filen_ame_length`みたいな名前が良さそうです。

JunichiIto commented on 05.ls/ls.rb:
```diff
@@ -0,0 +1,105 @@
+
+  file_path = File.expand_path(ARGV[0] || '')
+
+  files = load_files(file_path, option_a, option_l, option_r)
```
`load_files` = ファイルをロードする、という言い回しは業界的には「ファイルの中身を読み込む」と解釈されます。
loadは使わずに`find_files`とか`find_file_names`とかの方がよさそうです。

JunichiIto commented on 05.ls/ls.rb:
```diff
@@ -0,0 +1,105 @@
+
+def print_file_details(files, file_path)
+  sum_blocks = 0
+  file_details = files.map do |item|
```
複数形.map do |単数形| の形が良いので、`item`は`file`が良さそうです。

JunichiIto reviewed (commented):
名前付けについていくつかコメントしました〜。

Nabegon committed:
rename

max_length_string -> max_length, load_files -> find_files, files.map do |item|-> files.map do |file|

JunichiIto reviewed (approved):
確認しました。OKです！ 🙆‍♂️
